src/btree/bt_read.c

```diff
@@ -247,11 +247,35 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
                 return (WT_NOTFOUND);
             goto read;
         case WT_REF_DISK:
-            /* Limit reads to cache-only, or internal pages only. */
-            if (LF_ISSET(WT_READ_CACHE) ||
-              (LF_ISSET(WT_READ_CACHE_LEAF) && __wt_ref_is_leaf(session, ref)))
+            /* Limit reads to cache-only. */
+            if (LF_ISSET(WT_READ_CACHE))
                 return (WT_NOTFOUND);
 
+            /* Limit reads to internal pages only. */
+            if (LF_ISSET(WT_READ_CACHE_LEAF)) {
+                /*
+                 * Currently, the internal page read request is passed only in two scenarios.
+                 *  1. Garbage collection of history store
+                 *  2. Rollback to stable operation
+                 *
+                 * Lock the ref before we check the page type to avoid the ref getting changed
+                 * underneath. Failed to change the ref state to WT_REF_LOCKED can led skip the page
+                 * read.
```

Comment:
This sentence is confusing to me, it feels like there are some words missing: "Failed to change the ref state to WT_REF_LOCKED can led skip the page read." Maybe reword as "Failure to change the ref state to WT_REF_LOCKED can lead to skipping the page read" (if that is the correct interpretation!)

Reply:
Fixed with the provided sentence.